src/components/sign-attributes/sign-attribute-pane.vue

```diff
@@ -0,0 +1,259 @@
+                            v-for="attrValue in attr.values"
+                            :key="attrValue.id"
+                            @click="onAddAttribute(attr, attrValue)"
+                        >{{attrValue.value}}</b-dropdown-item>
+                    </b-dropdown>
+                </b-dropdown>
+            </li>
+            <li
+                class="pr-2"
+                v-for="attribute in attributes"
+                :key="attribute.interpretationAttributeId"
+            >
+                <sign-attribute
+                    :attribute="attribute"
+                    @attribute-click="onAttributeClick(attribute)"
+                />
+            </li>
+        </ul>
+        <sign-attribute-modal />
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue, Emit } from 'vue-property-decorator';
+import { Artefact } from '@/models/artefact';
+import {
+    TextFragmentData,
+    TextFragment,
+    SignInterpretation,
+    ArtefactTextFragmentData,
+} from '@/models/text';
+import TextFragmentComponent from '@/components/text/text-fragment.vue';
+import { EditionInfo } from '@/models/edition';
+import { AttributeDTO, AttributeValueDTO, InterpretationAttributeDTO } from '@/dtos/sqe-dtos';
+import SignAttribute from './sign-attribute.vue';
+import SignAttributeModal from './sign-attribute-modal.vue';
+import { TextFragmentAttributeOperation } from '@/views/artefact-editor/operations';
+import { BDropdown, BvEvent } from 'bootstrap-vue';
+
+@Component({
+    name: 'sign-attribute-pane',
+    components: {
+        'sign-attribute-modal': SignAttributeModal,
+        'sign-attribute': SignAttribute,
+    },
+})
+export default class SignAttributePane extends Vue {
+    private keepOpen = false;
+    private attributesMenu: AttributeDTO[] = [];
+
+    public get artefactEditor() {
+        return this.$state.artefactEditor;
+    }
+    public get selectedSignsInterpretation(): SignInterpretation[] {
+        return this.artefactEditor.selectedSignsInterpretation;
+    }
+
+    private get attributesMetadata() {
+        return (
+            this.$state.editions.current?.attributeMetadata?.allAttributes || []
+        );
+    }
+
+    public get attributes(): InterpretationAttributeDTO[] {
+        let attributeValues: number[] = [];
+        let first = true;
+
+        if (!this.selectedSignsInterpretation.length) {
+            return [];
+        }
+
+        for (const si of this.selectedSignsInterpretation) {
+            const siValues = [
+                ...si.attributes.map((attr) => attr.attributeValueId),
+            ];
+            if (first) {
+                first = false;
+                attributeValues = siValues;
+            } else {
+                attributeValues = attributeValues.filter((val) =>
+                    siValues.includes(val)
+                );
+            }
+        }
+
+        const attributes = this.selectedSignsInterpretation[0].attributes.filter(
```

Comment:
I guess there is always at least one attribute, so we don't need to guard against the possibility that `this.selectedSignsInterpretation[0]` doesn't exist?

Reply:
It is checked a few lines up. Added a comment to make sure it's clear.